api/STRUCTURE.md

```diff
@@ -0,0 +1,84 @@
+# All of Us Workbench API Structure & Technologies
```

Comment:
A couple notes / thoughts on file naming. Feel free to ignore (though I think it does help a lot when we get these things right, since it makes it obvious where the next person who comes along should put content):

1. We have an existing directory, api/doc/, which contains some similar "here's how parts of the system work" Markdown files. It might be cleaner for this file to live over there, so there's one place where all of our backend-related team docs live.
2. In terms of file naming, I think all-caps should probably be reserved for truly standard filenames, e.g. README or INDEX. Structure as a name isn't bad, but I'm not sure it perfectly captures the intent. Spitballing a few other ideas in case any of them jump out to you as an improvement:
  - api/doc/overview.md
  - api/doc/index.md (this could eventually become a summary page of sorts, which links out to the more topic-specific docs we already have)
  - api/doc/tech-stack.md
  - api/doc/getting-started.md

Reply:
That directory was not on my radar at all. I agree that it makes sense to put this document there, but I'm a bit concerned that I'd just be round filing it. Maybe I'll point out this directory and this file in both the top level and API level READMEs.

Reply:
@gjuggler there's a ticket somewhere (or at least in my head) where we break down the main README into other sections and files. We need a navigation convention where everything there is linked from the top and/or intermediate sections, and link back in turn as a courtesy.

One annoyance that I think actually stops people from updating that file is that it's not in the `api` project. So maybe a top-of-the-backend readme makes sense.